src/Network/CurlResult.php

```diff
@@ -133,6 +133,11 @@ private function checkSuccess()
 	{
 		$this->isSuccess = ($this->returnCode >= 200 && $this->returnCode <= 299) || $this->errorNumber == 0;
```

Comment:
Please remove `|| $this->errorNumber == 0` from this line instead of adding the following code block. The end result is exactly the same, except there are 5 lines of code instead of one.

Reply:
I saw that the tests failed on my changes - and realized that only error codes higher or equal 400 are errors. (3xx are redirects)